theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -1701,126 +1702,137 @@ def test_conv2d_grad_wrt_weights(self):
+        self.kern_shape = [(6, 2, 3, 3), (6, 2, 5, 3), (4, 2, 3, 3), (4, 1, 3, 5)]
+        self.top_shape = [(5, 6, 3, 3), (4, 6, 3, 3), (3, 4, 3, 1), (2, 4, 5, 3)]
         self.filter_dilation = (1, 1)
-        self.mode = theano.Mode(optimizer=None)
+        self.ref_mode = 'FAST_RUN'
 
     def test_fwd(self):
-        img = np.random.random(self.img_shape).astype('float32')
-        kern = np.random.random(self.kern_shape).astype('float32')
-        split_imgs = np.split(img, self.num_groups, axis=1)
-        split_kern = np.split(kern, self.num_groups, axis=0)
-        img_var = theano.tensor.tensor4('img')
-        kern_var = theano.tensor.tensor4('kern')
-
-        grouped_abstractconv_func = theano.tensor.nnet.abstract_conv.AbstractConv2d(imshp=None,
-                                                                                    kshp=None,
-                                                                                    border_mode=self.border_mode,
-                                                                                    subsample=self.subsample,
-                                                                                    filter_dilation=self.filter_dilation,
-                                                                                    num_groups=self.num_groups)
-        grouped_conv_output = grouped_abstractconv_func(img_var, kern_var)
-        grouped_func = theano.function([img_var, kern_var], grouped_conv_output, mode=self.mode)
-        grouped_output = grouped_func(img, kern)
-
-        normal_conv_output = conv2d_corr(img_var,
-                                         kern_var,
-                                         border_mode=self.border_mode,
-                                         subsample=self.subsample,
-                                         filter_dilation=self.filter_dilation)
-        normal_func = theano.function([img_var, kern_var], normal_conv_output,
-                                      mode='FAST_RUN')
-        normal_concat_output = [normal_func(img_arr, kern_arr)
-                                for img_arr, kern_arr in zip(split_imgs, split_kern)]
-        normal_concat_output = np.concatenate(normal_concat_output, axis=1)
-
-        utt.assert_allclose(grouped_output, normal_concat_output)
-
-        utt.verify_grad(grouped_abstractconv_func,
-                        [img, kern],
-                        mode=self.mode)
+        for imshp, kshp, groups in zip(self.img_shape, self.kern_shape, self.num_groups):
+            img = np.random.random(imshp).astype('float32')
+            kern = np.random.random(kshp).astype('float32')
+            split_imgs = np.split(img, groups, axis=1)
+            split_kern = np.split(kern, groups, axis=0)
+            img_sym = theano.tensor.tensor4('img')
+            kern_sym = theano.tensor.tensor4('kern')
+
+            grouped_abstractconv_func = self.conv2d(border_mode=self.border_mode,
```

Comment:
This isn't a func, but an op. You can change the variable name to just op. with a lower case, we use it elsewhere to mean an op instance.